launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -64,6 +64,10 @@
   /** Configuration key for the number of executor CPU cores. */
   public static final String EXECUTOR_CORES = "spark.executor.cores";
 
+  public static final String PYSPARK_DRIVER_PYTHON = "spark.pyspark.driver.python";
```

Comment:
This is not really what I meant by constants. I meant the `core/src/main/scala/org/apache/spark/internal/config/package.scala` file. Then your code to read the options would be:

```
conf.get(PYSPARK_DRIVER_PYTHON)
  .orElse(conf.get(PYSPATK_PYTHON))
  .orElse(sys.env.get("PYSPARK_DRIVER_PYTHON"))
  .getOrElse(sys.env.getOrElse("PYSPARK_PYTHON", "python"))
```

It would also be nice to update "configuration.md" with these new configs.